### ISSUE TITLE
Cloudformation failover configuratoin not fully configuring LTMs.

### ISSUE THREAD

**Issue by daveofthedogs** [F5Networks/f5-aws-cloudformation#63]
## Description
Describe the problem you're having or the enhancement you'd like to request.
Starting 1/14, when creating an 3-nic failover across AZs stack, the LTMs seem to get created and licenses (PAYG); however, the internal configuration of VLANs, self-ips, partitions, routes, device group members, and device gorups for HA is not completing.

I performed the same on Sunday, 1/13, and things seemed fine.  

I have tried v 4.1.1 and v4.1.0.  The pair I created Sunday was v4.1.0.

Also, looking for /var/log/cloud/aws/install.log to include, but it seems to not be on the LTMs.

## Template
For bugs, enter the template with which you are experiencing issues below.
f5-existing-stack-across-az-cluster-payg-3nic-bigip.template

## Severity Level
For bugs, enter the bug severity level. Do not set any labels.

Severity: 2

Severity level definitions:
 1. **Severity 1 (Critical)** : Defect is causing systems to be offline and/or nonfunctional. immediate attention is required.
 2. **Severity 2 (High)** : Defect is causing major obstruction of system operations.
 3. **Severity 3 (Medium)** : Defect is causing intermittent errors in system operations.
 4. **Severity 4 (Low)** : Defect is causing infrequent interuptions in system operations.
 5. **Severity 5 (Trival)** : Defect is not causing any interuptions to system operations, but none-the-less is a bug.

**Comment by daveofthedogs** [comment]
😢  Manual/guided configuration ensues

![2019-01-15_21-39-13](https://user-images.githubusercontent.com/783704/51222997-1095cc80-190e-11e9-993d-35781f694598.png)

**Comment by edwoodjrjr** [comment]
Hi daveofthedogs, 
TMK, nothing changed between Sunday and Monday.  Can you provide the following info for us to troubleshoot further?

- Did both versions fail on Monday?  Or just 4.1.1?
- Can you run "ls -l" on /config/cloud/ on the BIG-IP and post the output?  Want to make sure that all the required libraries downloaded correctly from github.
- Can you run "ifconfig" on the BIG-IP and post the output?  We have seen intermittent issues with AWS networking not being ready causing interfaces to be missing.

thanks

**Comment by crosbygw** [comment]
daveofthedogs, can you also share /var/log/cfn-init.log? Would be good to see if any errors occurred during cfn-init run time.

**Comment by daveofthedogs** [comment]
I will supply these in a bit. Just an FYI, I stood up 5 new pairs today, all of which did not complete network/HA config.  (I am getting very good and fast at manual configuration.)

**Comment by edwoodjrjr** [comment]
Okay.  I just ran a test and it did complete onboarding and network, and the HA config was correct on device 1 but the across_az_failover_group device group doesn't exist on device 2, so the sync state is disconnected.

**Comment by daveofthedogs** [comment]
Okay. Uploading the cfn-init.log. Obviously, there were network issues.  Not sure why, as the network is fine when I do the manual config.

Second aha is there is no /config/cloud.  That would be a huge issue.

CF set up these, and it seems both versions 5.1.0 and 4.1.1 had the problem.: F5 Networks BIGIP-13.1.1-0.0.4 PAYG - Good 5Gbps - 20180720201802-5d902121-d100-4e76-8d93-88b460373a93-ami-4a999e35.4 (ami-79557001)

[cfn-init.log](https://github.com/F5Networks/f5-aws-cloudformation/files/2769252/cfn-init.log)

**Comment by edwoodjrjr** [comment]
I believe this is related to our internal issue 854, for which we have a proposed fix.  Can you post the output of ifconfig here so we can verify?

**Comment by daveofthedogs** [comment]
`[dhorowit@load-api-router-4:Active:In Sync] ~ # ifconfig
eth0      Link encap:Ethernet  HWaddr 02:28:72:45:F8:DA
          inet6 addr: fe80::28:72ff:fe45:f8da/64 Scope:Link
          UP BROADCAST RUNNING ALLMULTI MULTICAST  MTU:1500  Metric:1
          RX packets:240535 errors:0 dropped:0 overruns:0 frame:0
          TX packets:246266 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:113018049 (107.7 MiB)  TX bytes:115664071 (110.3 MiB)

eth1      Link encap:Ethernet  HWaddr 02:EB:4E:68:91:9A
          inet6 addr: fe80::eb:4eff:fe68:919a/64 Scope:Link
          UP BROADCAST RUNNING PROMISC ALLMULTI MULTICAST  MTU:1500  Metric:1
          RX packets:56273 errors:0 dropped:0 overruns:0 frame:0
          TX packets:89923 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:7704701 (7.3 MiB)  TX bytes:5457871 (5.2 MiB)

eth2      Link encap:Ethernet  HWaddr 02:03:77:2F:F8:BA
          inet6 addr: fe80::3:77ff:fe2f:f8ba/64 Scope:Link
          UP BROADCAST RUNNING PROMISC ALLMULTI MULTICAST  MTU:1500  Metric:1
          RX packets:925153 errors:0 dropped:0 overruns:0 frame:0
          TX packets:961328 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:228289354 (217.7 MiB)  TX bytes:238355358 (227.3 MiB)

external  Link encap:Ethernet  HWaddr 02:EB:4E:68:91:9A
          inet addr:10.156.115.120  Bcast:10.156.115.127  Mask:255.255.255.192
          inet6 addr: fe80::eb:4eff:fe68:919a/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:39948 errors:0 dropped:0 overruns:0 frame:0
          TX packets:74524 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:3283402 (3.1 MiB)  TX bytes:4171279 (3.9 MiB)

internal  Link encap:Ethernet  HWaddr 02:03:77:2F:F8:BA
          inet addr:10.156.113.29  Bcast:10.156.113.255  Mask:255.255.255.0
          inet6 addr: fe80::3:77ff:fe2f:f8ba/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:740458 errors:0 dropped:0 overruns:0 frame:0
          TX packets:736848 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:171221722 (163.2 MiB)  TX bytes:176075848 (167.9 MiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.255.255.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:3218826 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3218826 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:731867318 (697.9 MiB)  TX bytes:731867318 (697.9 MiB)

lo:1      Link encap:Local Loopback
          inet addr:127.2.0.2  Mask:255.255.255.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1

mgmt      Link encap:Ethernet  HWaddr 02:28:72:45:F8:DA
          inet addr:10.156.115.250  Bcast:10.156.115.255  Mask:255.255.255.224
          inet6 addr: fe80::28:72ff:fe45:f8da/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:240492 errors:0 dropped:0 overruns:0 frame:0
          TX packets:246271 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:109648323 (104.5 MiB)  TX bytes:115665477 (110.3 MiB)

tmm       Link encap:Ethernet  HWaddr 00:98:76:54:32:10
          inet addr:127.1.1.254  Bcast:127.1.1.255  Mask:255.255.255.0
          inet6 addr: fe80::298:76ff:fe54:3210/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:9282  Metric:1
          RX packets:270657 errors:0 dropped:0 overruns:0 frame:0
          TX packets:278498 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:14674032 (13.9 MiB)  TX bytes:40986650 (39.0 MiB)

tmm_bp    Link encap:Ethernet  HWaddr 02:01:23:45:00:FE
          inet addr:127.20.0.254  Bcast:127.20.255.255  Mask:255.255.0.0
          inet6 addr: fe80::1:23ff:fe45:fe/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1640  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:0 (0.0 b)  TX bytes:0 (0.0 b)`

**Comment by edwoodjrjr** [comment]
That is strange.  Not sure why your eth0, eth1, and eth2 are IPv6 only.  In my test I only have eth0 that's IPv6, but it's not present in the console or the metadata lookup we do to set routes, so that part would definitely fail.

It's not clear yet what's going on, but the planned fix includes getting the information we need from tmsh and not AWS metadata, so I think that would still solve this issue.

**Comment by daveofthedogs** [comment]
Any timeframe?  I may need to build 8 more in our prod env next week.   THe above are all our performance env.

**Comment by edwoodjrjr** [comment]
Will discuss with the team and let you know.

**Comment by daveofthedogs** [comment]
Interestingly, I created four failover clusters, and the setup seemd to have worked.  

```
admin@ip-10-156-111-155:Active:In Sync] ~ # ifconfig
eth0      Link encap:Ethernet  HWaddr 06:0E:AA:4A:4C:C0
          inet6 addr: fe80::40e:aaff:fe4a:4cc0/64 Scope:Link
          UP BROADCAST RUNNING ALLMULTI MULTICAST  MTU:9001  Metric:1
          RX packets:34344 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3842 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:49991563 (47.6 MiB)  TX bytes:411914 (402.2 KiB)

eth1      Link encap:Ethernet  HWaddr 06:85:1F:3E:92:1A
          inet6 addr: fe80::485:1fff:fe3e:921a/64 Scope:Link
          UP BROADCAST RUNNING PROMISC ALLMULTI MULTICAST  MTU:1500  Metric:1
          RX packets:12429 errors:0 dropped:0 overruns:0 frame:0
          TX packets:980 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:18259184 (17.4 MiB)  TX bytes:82429 (80.4 KiB)

eth2      Link encap:Ethernet  HWaddr 06:60:A4:80:CE:8E
          inet6 addr: fe80::460:a4ff:fe80:ce8e/64 Scope:Link
          UP BROADCAST RUNNING PROMISC ALLMULTI MULTICAST  MTU:1500  Metric:1
          RX packets:46074 errors:0 dropped:0 overruns:0 frame:0
          TX packets:48800 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:9592046 (9.1 MiB)  TX bytes:19821181 (18.9 MiB)

external  Link encap:Ethernet  HWaddr 06:85:1F:3E:92:1A
          inet addr:10.156.110.29  Bcast:10.156.110.127  Mask:255.255.255.128
          inet6 addr: fe80::485:1fff:fe3e:921a/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:2149 errors:0 dropped:0 overruns:0 frame:0
          TX packets:945 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:17704340 (16.8 MiB)  TX bytes:80379 (78.4 KiB)

internal  Link encap:Ethernet  HWaddr 06:60:A4:80:CE:8E
          inet addr:10.156.105.211  Bcast:10.156.105.255  Mask:255.255.254.0
          inet6 addr: fe80::460:a4ff:fe80:ce8e/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:43293 errors:0 dropped:0 overruns:0 frame:0
          TX packets:43121 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:9508146 (9.0 MiB)  TX bytes:19532333 (18.6 MiB)

lo        Link encap:Local Loopback
          inet addr:127.0.0.1  Mask:255.255.255.0
          inet6 addr: ::1/128 Scope:Host
          UP LOOPBACK RUNNING  MTU:65536  Metric:1
          RX packets:25275 errors:0 dropped:0 overruns:0 frame:0
          TX packets:25275 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:5226401 (4.9 MiB)  TX bytes:5226401 (4.9 MiB)

lo:1      Link encap:Local Loopback
          inet addr:127.2.0.2  Mask:255.255.255.0
          UP LOOPBACK RUNNING  MTU:65536  Metric:1

mgmt      Link encap:Ethernet  HWaddr 06:0E:AA:4A:4C:C0
          inet addr:10.156.111.155  Bcast:10.156.111.191  Mask:255.255.255.192
          inet6 addr: fe80::40e:aaff:fe4a:4cc0/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:9001  Metric:1
          RX packets:4829 errors:0 dropped:0 overruns:0 frame:0
          TX packets:3847 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:47976123 (45.7 MiB)  TX bytes:413568 (403.8 KiB)

tmm       Link encap:Ethernet  HWaddr 00:98:76:54:32:10
          inet addr:127.1.1.254  Bcast:127.1.1.255  Mask:255.255.255.0
          inet6 addr: fe80::298:76ff:fe54:3210/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:9282  Metric:1
          RX packets:24945 errors:0 dropped:0 overruns:0 frame:0
          TX packets:33272 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:1371110 (1.3 MiB)  TX bytes:27279838 (26.0 MiB)

tmm_bp    Link encap:Ethernet  HWaddr 02:01:23:45:00:FE
          inet addr:127.20.0.254  Bcast:127.20.255.255  Mask:255.255.0.0
          inet6 addr: fe80::1:23ff:fe45:fe/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:0 errors:0 dropped:0 overruns:0 frame:0
          TX packets:0 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1
          RX bytes:0 (0.0 b)  TX bytes:0 (0.0 b)
```

**Comment by edwoodjrjr** [comment]
Great.  Is the HA pair fully sync'ed?  The bug I fixed causes the cluster to sometimes be in a disconnected state.

**Comment by crosbygw** [comment]
Issue resolved in release [v4.1.2](https://github.com/F5Networks/f5-aws-cloudformation/releases/tag/v4.1.2)
